src/briefcase/integrations/base.py

```diff
@@ -71,9 +105,16 @@ def __init__(
             )
         )
 
-        # Built-in tools without any external dependencies
-        Subprocess.verify(tools=self)
-        Download.verify(tools=self)
+    def add_tool(self, name: str, tool: TOOL_T) -> TOOL_T:
+        """Add a tool to the cache.
+
+        :param name: Name of the tool to add to the cache; The tool will be
+            accessible as an attribute of the ToolCache using this name.
+        :param tool: The instantiated Tool to make available.
+        :return: The added Tool.
+        """
+        setattr(self, name, tool)
+        return tool
```

Comment:
This strikes me as an odd construction - is there a reason for using this instead of the previous `tools.mytool = mytool; return mytool`? Unless we're planning to do additional verification on tools when they're added, I'm not sure I see what we gain over simple attribute assignment.

Reply:
This is just for evaluating things added as tools during testing so every tool could be confirmed as typed in `ToolCache`. It doesn't serve a functional purpose otherwise.

Reply:
I'm not sure I follow how that validation works (or how it's a guarantee) - could you elaborate? 

On top of that - my concern would be that `add_tool` is sufficiently un-idiomatic that the next tool added will inadvertently use `tool.mytool = ...`, and as a result, won't get picked up in testing.

Reply:
I created `ProxyToolCache` in `tests.integrations.conftest` to inspect if tools added through `ToolCache.add_tool()` are annotated in `ToolCache`.

I agree it is straightforward to skip using this wrapper altogether; it would ultimately depend on devs and the reviewers enforcing it.

For the record, I'm not a big fan either of this either given the architecture of `ToolCache`....I just couldn't think of anything else to try to meet your reasonable comment below:

> looking forward, if a new tool is added, there's nothing other than goodwill to ensure that any additional type declarations are added.